src/lib/Libutil/hook.c

```diff
@@ -275,6 +277,13 @@ hook_event_as_string(unsigned int event)
 		ev_ct++;
 	}
 
+	if (event & HOOK_EVENT_EXECJOB_ABORT) {
+		if (ev_ct > 0)
+			strncat(eventstr, ",", sizeof(eventstr) - strlen(eventstr) - 1);
+		strncat(eventstr, HOOKSTR_EXECJOB_ABORT, sizeof(eventstr) - strlen(eventstr) - 1);
+		ev_ct++;
+	}
+
```

Comment:
This if block is repeated many times and can be refactored using a loop.

Reply:
Wouldn't you need the HOOKSTR_* macros to be entries in a corresponding char* array? How will you correlate between the actual hook event and the hook string?

Reply:
Yes we will need to create a char* array for the `HOOKSTR`s and assign index to each hook event.

Reply:
I don't disagree with you @minghui-liu, but it does broaden the scope of the changes. I think @bayucan should have the option to file a ticket and address this later.

Reply:
@minghui-liu: I agree with you also on the refactoring, but I would like to work on it separate from this PR. I will file an issue to refactor code in hook.c to be worked on later.

Reply:
@bayucan - Please link the new ticket here and mark the conversation as resolved.

Reply:
Filed PP-1334: Refactor Libutil/hook.c code to eliminate repetitive if-else blocks.